frontend/src/pages/projects/screens/detail/notebooks/NotebookStorageBars.tsx

```diff
@@ -1,34 +1,41 @@
 import * as React from 'react';
 import {
   Alert,
-  Button,
+  Flex,
+  FlexItem,
   List,
   ListItem,
   Spinner,
   Stack,
   StackItem,
   Text,
 } from '@patternfly/react-core';
-import { PlusCircleIcon } from '@patternfly/react-icons';
+
 import { NotebookKind } from '~/k8sTypes';
 import useNotebookPVCItems from '~/pages/projects/pvc/useNotebookPVCItems';
 import StorageSizeBar from '~/pages/projects/components/StorageSizeBars';
 import { getNotebookPVCMountPathMap } from '~/pages/projects/notebook/utils';
 import { getDisplayNameFromK8sResource } from '~/concepts/k8s/utils';
+import { ProjectObjectType } from '~/concepts/design/utils';
+import HeaderIcon from '~/concepts/design/HeaderIcon';
+import InlineTruncatedClipboardCopy from '~/components/InlineTruncatedClipboardCopy';
+import ShowAllButton from './ShowAllButton';
 
 type NotebookStorageBarsProps = {
   notebook: NotebookKind;
-  onAddStorage: (notebook: NotebookKind) => void;
 };
 
-const NotebookStorageBars: React.FC<NotebookStorageBarsProps> = ({ notebook, onAddStorage }) => {
+const NotebookStorageBars: React.FC<NotebookStorageBarsProps> = ({ notebook }) => {
   const [pvcs, loaded, loadError] = useNotebookPVCItems(notebook);
   const mountFolderMap = getNotebookPVCMountPathMap(notebook);
+  const [showAll, setShowAll] = React.useState(false);
+  const defaultVisibleLength = 2;
+  const visiblePvcs = showAll ? pvcs : pvcs.slice(0, defaultVisibleLength);
 
   return (
     <Stack>
```

Comment:
I think you will need a `hasGutter` here after adding the show all button to make the layout look better.